### PR TITLE
qt5: add option to build Qt5 statically

### PR DESCRIPTION
**Pull request by sapiippo**
Add PACKAGECONFIG that enables static build for qtbase and consequently
to all Qt recipes. This can be enabled with DISTRO_FEATURE 'qt5-static',
which also adds qtdeclarative-native dependency to all recipes that
depend on qtdeclarative as it's required for qmlimportscannertool.

Building examples and tests with static build can take excessive amounts
of time and disk space, so disabling DISTRO_FEATURE 'ptest' and
PACKAGECONFIGs 'examples' from qtbase is advised.

Not all recipes support static builds and those are not fixed here.

Signed-off-by: Samuli Piippo <samuli.piippo@qt.io>

### CODE REVIEW

File: recipes-qt/qt5/qtbase/0001-Add-linux-oe-g-platform.patch [hunk 1]
```diff
@@ -63,6 +64,30 @@ index 934a18a924..0f5b1b6333 100644
...
+diff --git a/mkspecs/features/qt.prf b/mkspecs/features/qt.prf
+index b57afcf72d..afa1c39b3e 100644
+--- a/mkspecs/features/qt.prf
++++ b/mkspecs/features/qt.prf
+@@ -147,7 +147,7 @@ import_plugins:qtConfig(static) {
+         !isEmpty(plug_type) {
+             plug_path = $$eval(QT_PLUGIN.$${plug}.PATH)
+             isEmpty(plug_path): \
+-                plug_path = $$[QT_INSTALL_PLUGINS/get]
++                plug_path = $$[QT_INSTALL_PLUGINS]
```
Comment (otavio) [inline recipes-qt/qt5/qtbase/0001-Add-linux-oe-g-platform.patch]: Is this change related?

File: recipes-qt/qt5/qtbase/0001-Add-linux-oe-g-platform.patch [hunk 2]
```diff
@@ -63,6 +64,30 @@ index 934a18a924..0f5b1b6333 100644
...
+             LIBS += -L$$plug_path/$$plug_type
+         }
+         LIBS += -l$${plug}$$qtPlatformTargetSuffix()
+@@ -298,8 +298,8 @@ for(ever) {
+ # static builds: link qml import plugins into the target.
+ contains(all_qt_module_deps, qml): \
+         qtConfig(static):import_plugins:!host_build:!no_import_scan {
+-    exists($$[QT_INSTALL_QML/get]): \
+-        QMLPATHS *= $$[QT_INSTALL_QML/get]
```
Comment (otavio) [inline recipes-qt/qt5/qtbase/0001-Add-linux-oe-g-platform.patch]: Same here.

**Comment by sapiippo** [comment]
Both are relevant changes, otherwise qmake will use paths to the recipe-sysroot-native

**Comment by sapiippo** [comment]
No side effects, since both sections are guarded with static: (or with QtConfig(static): ), so they are not used unless you are doing static build.

File: recipes-qt/qt5/qtdeclarative_git.bb [hunk 3]
```diff
@@ -12,7 +12,7 @@ LIC_FILES_CHKSUM = " \
     file://LICENSE.FDL;md5=6d9f2a9af4c8b8c3c769f6cc1b6aaf7e \
 "
 
-DEPENDS += "qtbase"
+DEPENDS += "qtbase ${@bb.utils.contains('DISTRO_FEATURES', 'qt5-static', 'qtdeclarative-native', '', d)}"
```
Comment (otavio) [inline recipes-qt/qt5/qtdeclarative_git.bb]: Please add a PACKAGECONFIG[static].
